velox/expression/tests/CastExprTest.cpp

```diff
@@ -1992,6 +1992,106 @@ TEST_F(CastExprTest, castInTry) {
       ARRAY(ARRAY(VARCHAR())), ARRAY(ARRAY(BIGINT())), nested, nestedExpected);
 }
 
+TEST_F(CastExprTest, doubleToDecimal) {
+  // Double to short decimal.
+  const auto input =
+      makeFlatVector<double>({-3333.03, -2222.02, -1.0, 0.00, 100, 99999.99});
+  testComplexCast(
+      "c0",
+      input,
+      makeFlatVector<int64_t>(
+          {-33'330'300, -22'220'200, -10'000, 0, 1'000'000, 999'999'900},
+          DECIMAL(10, 4)));
+
+  // Double to long decimal.
+  testComplexCast(
```

Comment:
Can you remind me what does 'complex' mean here? This is not about complex types, is it?

Reply:
It is similar with testCast with some tiny differences. This is one PR https://github.com/facebookincubator/velox/pull/8254 to combine them inspired by https://github.com/facebookincubator/velox/pull/7377#discussion_r1440093802. cc @kagamiori

Reply:
Got it. Thank you for clarifying. Let's review and land that other PR to reduce confusion.